crates/sui-graphql-rpc/schema/current_progress_schema.graphql

```diff
@@ -2001,6 +2001,18 @@ type ServiceConfig {
 	"""
 	maxQueryNodes: Int!
 	"""
+	The maximum number of output nodes in a GraphQL response.
+	
+	If a node is a connection, it is counted as the specified 'first' or 'last' number of items,
+	or the default_page_size as set by the server. Non-connection nodes and fields are not included
+	in this count.
```

Comment:
This bit is not true -- non-connection nodes and fields are counted, as 1, so if you have a connection with first 10, and within that you request three non-connection fields, we will calculate the query as costing 3 * 10 = 30 output nodes.

Reply:
Got it, this is a slight deviation from how something like github would calculate the total nodes in a call. Seems like there's overlap with the existing MAX_QUERY_NODES today, only that validation considers all fields of cost 1 instead of N if is a connection.

Reply:
I still lean towards not counting non-connection fields as nodes though. Fields like address, digest, etc. are moreso data contained within the node rather than individual nodes themselves. We have some exceptions, like object.previousTransactionBlock or total_checkpoints, and we can make our output node estimation smarter to account for these as well, but I'm inclined to think that limiting nodes to just connection fields would make the MAX_OUTPUT_NODES value easier to reason about, as they're the main contributor to the additional nodes we fetch. Otherwise, I think counting nodes in this calculation would increase the perceived complexity of the query without necessarily reflecting the actual complexity.

We also have MAX_QUERY_NODES and MAX_QUERY_DEPTH which limit the total number of fields and depth already, so there would be some overlap if MAX_OUTPUT_NODES also considers those in its calculation.

But this is not a hard stance, and happy to be swayed otherwise. We also have existing implementations for MAX_QUERY_NODES and MAX_QUERY_DEPTH, so we can just plug in there for MAX_OUTPUT_NODES

Reply:
I touched on this in the other PR, but what you're getting at here:

> Fields like address, digest, etc. are moreso data contained within the node rather than individual nodes themselves.

Is reasonable, and is the distinction between scalar and non-scalar values.  I would be OK with applying the limit only to non-scalar nodes, because these most directly map to things we need to fetch from the DB, and luckily, you can figure out statically whether a value is scalar or not, because if it's non-scalar, it must have a selection-set.